pkg/controller/daemon/daemon_controller.go

```diff
@@ -682,8 +682,8 @@ func (dsc *DaemonSetsController) getNodesToDaemonPods(ds *extensions.DaemonSet)
 	// Group Pods by Node name.
 	nodeToDaemonPods := make(map[string][]*v1.Pod)
 	for _, pod := range claimedPods {
-		// Skip terminating pods
-		if pod.DeletionTimestamp != nil {
+		// Skip terminated pods, consider terminating pods ie: pods serving terminationGracePeriod
+		if pod.DeletionTimestamp != nil && pod.DeletionTimestamp.Before(metav1.Now()) {
```

Comment:
I don't think this does what you want it to. This is going to regress #44144. The controller will consider adopting Pods that deleted Pods.

Reply:
#51279 is trying to fix that same issue

Reply:
Will close this one